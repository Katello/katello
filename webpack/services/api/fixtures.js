export default [
  {
    path: '/katello/api/v2/products/:id/repository_sets/:repo_set_id/enable',
    searchRegex: /\/katello\/api\/v2\/products\/\d+\/repository_sets\/\d+\/enable/,
    type: 'PUT',
    response: () => {
      const types = ['yum', 'source_rpm', 'debug', 'iso', 'beta', 'kickstart'];

      const output = {
        id: Math.round(Math.random() * 500),
        name: 'Red Hat Enterprise Linux 6.2 Server for RHS 2 VSA Beta RPMs from RHUI x86_64',
        content_type: types[Math.floor(Math.random() * types.length)],
      };

      return [
        200,
        {
          id: 'd4a4432e-4987-4206-aa99-615b5b15fbf8',
          label: 'Actions::Katello::RepositorySet::EnableRepository',
          pending: false,
          username: 'admin',
          started_at: '2017-12-19 14:33:41 UTC',
          ended_at: '2017-12-19 14:33:44 UTC',
          state: 'stopped',
          result: 'success',
          progress: 1.0,
          input: {
            services_checked: ['pulp', 'pulp_auth', 'candlepin', 'candlepin_auth'],
            repository: {
              id: 7,
              name: 'Red Hat Enterprise Linux 6.2 Server for RHS 2 VSA Beta RPMs from RHUI x86_64',
              label: 'Red_Hat_Enterprise_Linux_6_2_Server_for_RHS_2_VSA_Beta_RPMs_from_RHUI_x86_64',
            },
            product: {
              id: 155,
              name: 'Red Hat Storage',
              label: 'Red_Hat_Storage',
              cp_id: '171',
            },
            provider: {
              id: 2,
              name: 'Red Hat',
            },
            organization: {
              id: 1,
              name: 'Default Organization',
              label: 'Default_Organization',
            },
          },
          output,
          humanized: {
            action: 'Enable',
            input: [],
            output: '',
            errors: [],
          },
          cli_example: null,
        },
      ];
    },
  },
  {
    path: '/katello/api/v2/products/:id/repository_sets/:repo_set_id/disable',
    searchRegex: /\/katello\/api\/v2\/products\/\d+\/repository_sets\/\d+\/disable/,
    type: 'PUT',
    response: () => [
      200,
      {
        id: 'd4a4432e-4987-4206-aa99-615b5b15fbf8',
        label: 'Actions::Katello::RepositorySet::DisableRepository',
        pending: false,
        username: 'admin',
        started_at: '2017-12-19 14:33:41 UTC',
        ended_at: '2017-12-19 14:33:44 UTC',
        state: 'stopped',
        result: 'success',
        progress: 1.0,
        input: {
          services_checked: ['pulp', 'pulp_auth', 'candlepin', 'candlepin_auth'],
          repository: {
            id: 7,
            name: 'Red Hat Enterprise Linux 6.2 Server for RHS 2 VSA Beta RPMs from RHUI x86_64',
            label: 'Red_Hat_Enterprise_Linux_6_2_Server_for_RHS_2_VSA_Beta_RPMs_from_RHUI_x86_64',
          },
          product: {
            id: 155,
            name: 'Red Hat Storage',
            label: 'Red_Hat_Storage',
            cp_id: '171',
          },
          provider: {
            id: 2,
            name: 'Red Hat',
          },
          organization: {
            id: 1,
            name: 'Default Organization',
            label: 'Default_Organization',
          },
        },
        output: {},
        humanized: {
          action: 'Disable',
          input: [],
          output: '',
          errors: [],
        },
        cli_example: null,
      },
    ],
  },
  {
    path: '/organizations/:id/repository_sets',
    searchRegex: /\/organizations\/\d+\/repository_sets$/,
    response: () => [
      200,
      {
        total: 5,
        subtotal: 5,
        page: null,
        per_page: null,
        error: null,
        search: null,
        sort: {
          by: null,
          order: null,
        },
        results: [
          {
            repositories: [],
            product: {
              name: 'Red hat Enterprise Linux Server 7',
              id: 5,
            },
            type: 'file',
            vendor: 'Red Hat',
            gpgUrl: 'http://',
            contentUrl: '/content/dist/rhel/server/7///source/iso',
            id: '2457',
            name: 'Red Hat Enterprise Linux 7 Server (Source ISOs)',
            label: 'rhel-7-server-source-isos',
          },
          {
            repositories: [],
            type: 'yum',
            product: {
              name: 'Red hat Enterprise Linux Server 7',
              id: 5,
            },
            vendor: 'Red Hat',
            gpgUrl: 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release',
            contentUrl: '/content/dist/rhel/server/6/6Server//sat-tools/6.1/source/SRPMS',
            id: '4187',
            name: 'Red Hat Satellite Tools 6.1 (for RHEL 6 Server) (Source RPMs)',
            label: 'rhel-6-server-satellite-tools-6.1-source-rpms',
          },
          {
            repositories: [],
            type: 'yum',
            product: {
              name: 'Red hat Enterprise Linux Server 7',
              id: 5,
            },
            vendor: 'Red Hat',
            gpgUrl:
              'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-beta,file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release',
            contentUrl: '/content/beta/rhel/server/6/6Server//rh-common/os',
            id: '3076',
            name: 'Red Hat Enterprise Linux 6 Server - RH Common Beta (RPMs)',
            label: 'rhel-6-server-rh-common-beta-rpms',
          },
          {
            repositories: [],
            type: 'yum',
            product: {
              name: 'Red hat Enterprise Linux Server 7',
              id: 5,
            },
            vendor: 'Red Hat',
            gpgUrl: 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release',
            contentUrl: '/content/dist/rhel/server/6///rhev-agent/3/os',
            id: '1699',
            name: 'Red Hat Enterprise Virtualization Agents for RHEL 6 Server (RPMs)',
            label: 'rhel-6-server-rhev-agent-rpms',
          },
          {
            repositories: [
              {
                id: 631,
                name: 'Red Hat Enterprise Linux 6 Server Kickstart x86_64 6.8',
                releasever: '6.8',
                arch: 'x86_64',
              },
            ],
            product: {
              name: 'Red hat Enterprise Linux Server 7',
              id: 5,
            },
            type: 'kickstart',
            vendor: 'Red Hat',
            gpgUrl: 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release',
            contentUrl: '/content/dist/rhel/server/6///kickstart',
            id: '1952',
            name: 'Red Hat Enterprise Linux 6 Server (Kickstart)',
            label: 'rhel-6-server-kickstart',
          },
        ],
      },
    ],
  },
  {
    // Returns the currently enabled repositories
    path: '/repository_sets?organization_id=:id',
    searchRegex: /\/organizations\/\d+\/repository_sets/,
    response: () => [
      200,
      {
        total: 5,
        subtotal: 5,
        page: null,
        per_page: null,
        error: null,
        search: null,
        sort: {
          by: null,
          order: null,
        },
        results: [
          {
            repositories: [
              {
                id: 631,
                name: 'Red Hat Enterprise Linux 6 Server Kickstart x86_64 6.8',
                releasever: '6.8',
                arch: 'x86_64',
              },
            ],
            product: {
              name: 'Red hat Enterprise Linux Server 7',
              id: 5,
            },
            type: 'kickstart',
            vendor: 'Red Hat',
            gpgUrl: 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release',
            contentUrl: '/content/dist/rhel/server/6///kickstart',
            id: '1952',
            name: 'Red Hat Enterprise Linux 6 Server (Kickstart)',
            label: 'rhel-6-server-kickstart',
          },
        ],
      },
    ],
  },
  {
    path: '/products/:product_id/repository_sets/:content_id/available_repositories',
    searchRegex: /\/products\/\d+\/repository_sets\/\d+\/available_repositories/,
    response: () => [
      200,
      {
        total: 6,
        subtotal: 6,
        page: null,
        per_page: null,
        error: null,
        search: null,
        sort: {
          by: null,
          order: null,
        },
        results: [
          {
            substitutions: {
              releasever: '7.0',
              basearch: 'x86_64',
            },
            path: '/content/dist/rhel/server/7/7.0/x86_64/supplementary/os',
            repo_name: 'Red Hat Enterprise Linux 7 Server - Supplementary RPMs x86_64 7.0',
            name: 'Red Hat Enterprise Linux 7 Server - Supplementary (RPMs)',
            pulp_id:
              'Default_Organization-Red_Hat_Enterprise_Linux_Server-Red_Hat_Enterprise_Linux_7_Server_-_Supplementary_RPMs_x86_64_7_0',
            enabled: false,
            promoted: false,
          },
          {
            substitutions: {
              releasever: '7.1',
              basearch: 'x86_64',
            },
            path: '/content/dist/rhel/server/7/7.1/x86_64/supplementary/os',
            repo_name: 'Red Hat Enterprise Linux 7 Server - Supplementary RPMs x86_64 7.1',
            name: 'Red Hat Enterprise Linux 7 Server - Supplementary (RPMs)',
            pulp_id:
              'Default_Organization-Red_Hat_Enterprise_Linux_Server-Red_Hat_Enterprise_Linux_7_Server_-_Supplementary_RPMs_x86_64_7_1',
            enabled: false,
            promoted: false,
          },
          {
            substitutions: {
              releasever: '7.2',
              basearch: 'x86_64',
            },
            path: '/content/dist/rhel/server/7/7.2/x86_64/supplementary/os',
            repo_name: 'Red Hat Enterprise Linux 7 Server - Supplementary RPMs x86_64 7.2',
            name: 'Red Hat Enterprise Linux 7 Server - Supplementary (RPMs)',
            pulp_id:
              'Default_Organization-Red_Hat_Enterprise_Linux_Server-Red_Hat_Enterprise_Linux_7_Server_-_Supplementary_RPMs_x86_64_7_2',
            enabled: false,
            promoted: false,
          },
          {
            substitutions: {
              releasever: '7.3',
              basearch: 'x86_64',
            },
            path: '/content/dist/rhel/server/7/7.3/x86_64/supplementary/os',
            repo_name: 'Red Hat Enterprise Linux 7 Server - Supplementary RPMs x86_64 7.3',
            name: 'Red Hat Enterprise Linux 7 Server - Supplementary (RPMs)',
            pulp_id:
              'Default_Organization-Red_Hat_Enterprise_Linux_Server-Red_Hat_Enterprise_Linux_7_Server_-_Supplementary_RPMs_x86_64_7_3',
            enabled: false,
            promoted: false,
          },
          {
            substitutions: {
              releasever: '7.4',
              basearch: 'x86_64',
            },
            path: '/content/dist/rhel/server/7/7.4/x86_64/supplementary/os',
            repo_name: 'Red Hat Enterprise Linux 7 Server - Supplementary RPMs x86_64 7.4',
            name: 'Red Hat Enterprise Linux 7 Server - Supplementary (RPMs)',
            pulp_id:
              'Default_Organization-Red_Hat_Enterprise_Linux_Server-Red_Hat_Enterprise_Linux_7_Server_-_Supplementary_RPMs_x86_64_7_4',
            enabled: false,
            promoted: false,
          },
          {
            substitutions: {
              releasever: '7Server',
              basearch: 'x86_64',
            },
            path: '/content/dist/rhel/server/7/7Server/x86_64/supplementary/os',
            repo_name: 'Red Hat Enterprise Linux 7 Server - Supplementary RPMs x86_64 7Server',
            name: 'Red Hat Enterprise Linux 7 Server - Supplementary (RPMs)',
            pulp_id:
              'Default_Organization-Red_Hat_Enterprise_Linux_Server-Red_Hat_Enterprise_Linux_7_Server_-_Supplementary_RPMs_x86_64_7Server',
            enabled: false,
            promoted: false,
          },
        ],
      },
    ],
  },
];
