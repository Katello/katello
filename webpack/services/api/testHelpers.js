
export const failureAction = (type, message = 'Request failed with status code 422') => (
  {
    type,
    payload: {
      messages: [message],
      result: new Error(message),
    },
  }
);

export const toastErrorAction = (message = 'Request failed with status code 422') => (
  {
    payload: {
      message: {
        message,
        sticky: true,
        type: 'error',
      },
    },
    type: 'TOASTS_ADD',
  }
);

export default {
  failureAction,
  toastErrorAction,
};
