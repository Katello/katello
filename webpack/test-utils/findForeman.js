const path = require('path');
const fs = require('fs');

const getForemanLocation = () => {
  const foremanLocations = ['./foreman', '../foreman', '../../foreman'];
  const notFound = 'Foreman directory cannot be found! This action requires Foreman to be present ' +
  'in either a parent, sibling, or child directory relative to the plugin.';
  const currentDir = process.cwd();
  let fullPath;

  foremanLocations.forEach((relativeForemanPath) => {
    const possibleForemanPath = path.join(currentDir, relativeForemanPath);
    if (fs.existsSync(possibleForemanPath)) fullPath = possibleForemanPath;
  });

  if (!fullPath) throw new Error(notFound);
  return fullPath;
};

// Get a subdirectory within Foreman
const getForemanRelativePath = (relativeForemanPath) => {
  const foremanLocation = getForemanLocation();
  const notFound = `Could not find ${relativeForemanPath} in ${foremanLocation}`;
  const foremanRelativePath = path.join(foremanLocation, relativeForemanPath);
  if (!fs.existsSync(foremanRelativePath)) throw new Error(notFound);
  return foremanRelativePath;
};

module.exports = { getForemanLocation, getForemanRelativePath };
