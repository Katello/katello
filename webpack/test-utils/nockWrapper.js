import nock from 'nock';

// Using the library 'nock' as it matches actual network requests rather than mock another
// library. This is helpful when the request is not coming from Katello. For example, axios
// called within Katello can be mocked with axios-mock-adapter or similar, but a http request
// made by axios that is coming from Foreman cannot be mocked by axios-mock-adapter or a
// jest mock within Katello. So to do this, we can mock the request a level deeper within
// nodejs by using nock.
export const nockInstance = nock('http://localhost');

// Calling .done() with nock asserts that the request was fufilled. We use a timeout to ensure
// that the component has set up and made the request before the assertion is made.
export const assertNockRequest = (scope, timeout = 10000) => {
  setTimeout(() => {
    scope.done();
  }, timeout);
};
