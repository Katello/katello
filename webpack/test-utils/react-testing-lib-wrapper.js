// For react-testing-library helpers, overrides, and utilities
// All elements from react-testing-library can be imported from this wrapper.
// See https://testing-library.com/docs/react-testing-library/setup for more info
import React from 'react';
import thunk from 'redux-thunk';
import Immutable from 'seamless-immutable';
import { APIMiddleware, reducers as apiReducer } from 'foremanReact/redux/API';
import { reducers as fillReducers } from 'foremanReact/components/common/Fill';
import { reducers as foremanModalReducer } from 'foremanReact/components/ForemanModal';
import { STATUS } from 'foremanReact/constants';
import { render, waitFor, waitForElementToBeRemoved } from '@testing-library/react';
import { createStore, applyMiddleware, combineReducers } from 'redux';
import { Provider } from 'react-redux';
import { MemoryRouter, BrowserRouter } from 'react-router-dom';
import { initialSettingsState } from '../scenes/Settings/SettingsReducer';
import allKatelloReducers from '../redux/reducers/index.js';

// r-t-lib's print limit for debug() is quite small, setting it to a much higher char max here.
// See https://github.com/testing-library/react-testing-library/issues/503 for more info.
process.env.DEBUG_PRINT_LIMIT = 99999;

// Renders testable component with redux and react-router according to Katello's usage
// This should be used when you want a fully connected component with Redux state and actions.
function renderWithRedux(
  component,
  {
    apiNamespace, // namespace if using API middleware
    initialApiState = { response: {}, status: STATUS.PENDING }, // Default state for API middleware
    initialState = {}, // Override full state
    routerParams = {},
  } = {},
) {
  // Adding the reducer in the expected namespaced format
  const combinedReducers = combineReducers({
    katello: allKatelloReducers,
    ...apiReducer,
    ...foremanModalReducer,
    ...fillReducers,
  });

  // Namespacing the initial state as well
  const initialFullState = Immutable({
    API: {
      [apiNamespace]: initialApiState,
    },
    katello: {
      settings: {
        settings: initialSettingsState,
      },
    },
    extendable: {},
    ...initialState,
  });
  const middlewares = applyMiddleware(thunk, APIMiddleware);
  const store = createStore(combinedReducers, initialFullState, middlewares);
  const connectedComponent = (
    <Provider store={store}>
      <MemoryRouter {...routerParams} >{component}</MemoryRouter>
    </Provider>
  );

  return { ...render(connectedComponent), store };
}

// When you actually need to change browser history
const renderWithRouter = (ui, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: BrowserRouter });
};

// When the tests run slower, they can hit the default waitFor timeout, which is 1000ms
// There doesn't seem to be a way to set it globally for r-t-lib, so using this wrapper function
const rtlTimeout = 5000;
export const patientlyWaitFor = waitForFunc => waitFor(waitForFunc, { timeout: rtlTimeout });
export const patientlyWaitForRemoval = waitForFunc =>
  waitForElementToBeRemoved(waitForFunc, { timeout: rtlTimeout });

// re-export everything, so the library can be used from this wrapper.
export * from '@testing-library/react';

export { renderWithRedux, renderWithRouter };
