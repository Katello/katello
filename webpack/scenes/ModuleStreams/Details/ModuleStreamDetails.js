import React, { Component } from 'react';
import BreadcrumbsBar from 'foremanReact/components/BreadcrumbBar';
import { translate as __ } from 'foremanReact/common/I18n';
import { PropTypes } from 'prop-types';
import api from '../../../services/api';
import ContentDetails from '../../../components/Content/Details/ContentDetails';
import moduleDetailsSchema from './ModuleDetailsSchema';

class ModuleStreamDetails extends Component {
  componentDidMount() {
    this.updateModuleStream();
  }

  componentDidUpdate(prevProps) {
    const { match: { params: prevRouterParams } } = this.props;
    const { match: { params: currentRouterParams } } = prevProps;
    if (prevRouterParams.id && (prevRouterParams.id !== currentRouterParams.id)) {
      this.updateModuleStream();
    }
  }

  updateModuleStream = () => {
    const moduleStreamId = parseInt(this.props.match.params.id, 10);
    this.props.loadModuleStreamDetails(moduleStreamId);
  };

  handleBreadcrumbSwitcherItem = (e, url) => {
    this.props.history.push(url);
    e.preventDefault();
  };

  render() {
    const { moduleStreamDetails } = this.props;
    const {
      loading, name, stream,
    } = moduleStreamDetails;

    const resource = {
      nameField: 'name',
      resourceUrl: api.getApiUrl('/module_streams'),
      switcherItemUrl: '/module_streams/:id',
    };

    return (
      <div>
        {!loading && <BreadcrumbsBar
          onSwitcherItemClick={(e, url) => this.handleBreadcrumbSwitcherItem(e, url)}
          data={{
            isSwitchable: true,
            breadcrumbItems: [
              {
                caption: __('Module Streams'),
                onClick: () =>
                  this.props.history.push('/module_streams'),
              },
              {
                caption: `${name} ${stream}`,
              },
            ],
            resource,
          }}
        />}
        <ContentDetails
          contentDetails={moduleStreamDetails}
          schema={moduleDetailsSchema(moduleStreamDetails)}
        />
      </div>
    );
  }
}

ModuleStreamDetails.propTypes = {
  loadModuleStreamDetails: PropTypes.func.isRequired,
  history: PropTypes.shape({ push: PropTypes.func.isRequired }).isRequired,
  location: PropTypes.shape({}).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
  moduleStreamDetails: PropTypes.shape({
    loading: PropTypes.bool,
    name: PropTypes.string,
    // Disabling rule as existing code failed due to an eslint-plugin-react update
    /* eslint-disable react/forbid-prop-types */
    profiles: PropTypes.array,
    repositories: PropTypes.array,
    artifacts: PropTypes.array,
    /* eslint-enable react/forbid-prop-types */
    stream: PropTypes.string,
  }).isRequired,
};

export default ModuleStreamDetails;
