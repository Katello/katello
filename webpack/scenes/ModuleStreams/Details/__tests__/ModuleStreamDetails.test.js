import ModuleStreamDetails from '../ModuleStreamDetails';
import { testComponentSnapshotsWithFixtures } from '../../../../move_to_pf/test-utils/testHelpers';
import { details, loadingState } from './moduleStreamDetails.fixtures';

jest.mock('foremanReact/components/Pagination/PaginationWrapper');

const mockFunc = jest.fn();

const baseProps = {
  loadModuleStreamDetails: mockFunc,
  location: { search: '' },
  history: { push: mockFunc },
  moduleStreamDetails: details,
  match: { params: { id: String(details.id) } },
};

const fixtures = {
  'renders with module stream provided': {
    ...baseProps,
  },
  'renders with loading state': {
    ...baseProps,
    moduleStreamDetails: loadingState,
  },
};

describe('Module stream details page', () => {
  describe('rendering', () => testComponentSnapshotsWithFixtures(ModuleStreamDetails, fixtures));
});
