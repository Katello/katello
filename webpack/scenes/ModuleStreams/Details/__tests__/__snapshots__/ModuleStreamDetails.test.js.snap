// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module stream details page rendering renders with loading state 1`] = `
<div>
  <LoadingState
    loading={true}
    loadingText="Loading"
    timeout={300}
  >
    <ForwardRef
      defaultActiveKey={1}
      id="module-stream-tabs-container"
    >
      <Grid
        bsClass="container-fluid"
        componentClass="div"
        fluid={false}
      >
        <Row
          bsClass="row"
          componentClass="div"
        >
          <Col
            bsClass="col"
            componentClass="div"
            sm={12}
          >
            <Nav
              bsClass="nav nav-tabs"
              justified={false}
              pullLeft={false}
              pullRight={false}
              stacked={false}
            >
              <NavItem
                active={false}
                disabled={false}
                eventKey={1}
              >
                <div>
                  Details
                </div>
              </NavItem>
              <NavItem
                active={false}
                disabled={false}
                eventKey={2}
              >
                <div>
                  Repositories
                </div>
              </NavItem>
              <NavItem
                active={false}
                disabled={false}
                eventKey={3}
              >
                <div>
                  Profiles
                </div>
              </NavItem>
              <NavItem
                active={false}
                disabled={false}
                eventKey={4}
              >
                <div>
                  Artifacts
                </div>
              </NavItem>
            </Nav>
          </Col>
        </Row>
        <TabContent
          animation={false}
          bsClass="tab"
          componentClass="div"
          mountOnEnter={false}
          unmountOnExit={false}
        >
          <TabPane
            bsClass="tab-pane"
            eventKey={1}
          >
            <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                sm={12}
              >
                <ModuleStreamDetailInfo
                  moduleStreamDetails={
                    Object {
                      "itemCount": 0,
                      "loading": true,
                      "pagination": Object {
                        "page": 0,
                        "perPage": 20,
                      },
                      "results": Array [],
                    }
                  }
                />
              </Col>
            </Row>
          </TabPane>
          <TabPane
            bsClass="tab-pane"
            eventKey={2}
          >
            <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                sm={12}
              >
                No repositories to show
              </Col>
            </Row>
          </TabPane>
          <TabPane
            bsClass="tab-pane"
            eventKey={3}
          >
            <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                sm={12}
              >
                No profiles to show
              </Col>
            </Row>
          </TabPane>
          <TabPane
            bsClass="tab-pane"
            eventKey={4}
          >
            <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                sm={12}
              >
                No artifacts to show
              </Col>
            </Row>
          </TabPane>
        </TabContent>
      </Grid>
    </ForwardRef>
  </LoadingState>
</div>
`;

exports[`Module stream details page rendering renders with module stream provided 1`] = `
<div>
  <BreadcrumbsBar
    data={
      Object {
        "breadcrumbItems": Array [
          Object {
            "caption": "Module Streams",
            "onClick": [Function],
          },
          Object {
            "caption": "avocado latest",
          },
        ],
        "isSwitchable": true,
        "resource": Object {
          "nameField": "name",
          "resourceUrl": "/katello/api/v2/module_streams",
          "switcherItemUrl": "/module_streams/:id",
        },
      }
    }
    onSwitcherItemClick={[Function]}
  />
  <LoadingState
    loading={false}
    loadingText="Loading"
    timeout={300}
  >
    <ForwardRef
      defaultActiveKey={1}
      id="module-stream-tabs-container"
    >
      <Grid
        bsClass="container-fluid"
        componentClass="div"
        fluid={false}
      >
        <Row
          bsClass="row"
          componentClass="div"
        >
          <Col
            bsClass="col"
            componentClass="div"
            sm={12}
          >
            <Nav
              bsClass="nav nav-tabs"
              justified={false}
              pullLeft={false}
              pullRight={false}
              stacked={false}
            >
              <NavItem
                active={false}
                disabled={false}
                eventKey={1}
              >
                <div>
                  Details
                </div>
              </NavItem>
              <NavItem
                active={false}
                disabled={false}
                eventKey={2}
              >
                <div>
                  Repositories
                </div>
              </NavItem>
              <NavItem
                active={false}
                disabled={false}
                eventKey={3}
              >
                <div>
                  Profiles
                </div>
              </NavItem>
              <NavItem
                active={false}
                disabled={false}
                eventKey={4}
              >
                <div>
                  Artifacts
                </div>
              </NavItem>
            </Nav>
          </Col>
        </Row>
        <TabContent
          animation={false}
          bsClass="tab"
          componentClass="div"
          mountOnEnter={false}
          unmountOnExit={false}
        >
          <TabPane
            bsClass="tab-pane"
            eventKey={1}
          >
            <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                sm={12}
              >
                <ModuleStreamDetailInfo
                  moduleStreamDetails={
                    Object {
                      "arch": "x86_64",
                      "artifacts": Array [
                        Object {
                          "id": 208,
                          "name": "python3-avocado-plugins-varianter-yaml-to-mux-0:63.0-2.module_2037+1b0ad681.noarch",
                        },
                        Object {
                          "id": 207,
                          "name": "python3-avocado-plugins-varianter-pict-0:63.0-2.module_2037+1b0ad681.noarch",
                        },
                      ],
                      "context": "a5b0195c",
                      "description": "Avocado is a set of tools and libraries (what people call these days a framework) to perform automated testing.",
                      "id": 22,
                      "name": "avocado",
                      "profiles": Array [
                        Object {
                          "id": 37,
                          "name": "default",
                          "rpms": Array [
                            Object {
                              "id": 108,
                              "name": "perl",
                            },
                            Object {
                              "id": 110,
                              "name": "foo",
                            },
                            Object {
                              "id": 111,
                              "name": "rpm_0",
                            },
                            Object {
                              "id": 112,
                              "name": "rpm_1",
                            },
                            Object {
                              "id": 113,
                              "name": "rpm_2",
                            },
                            Object {
                              "id": 114,
                              "name": "rpm_3",
                            },
                            Object {
                              "id": 115,
                              "name": "rpm_4",
                            },
                            Object {
                              "id": 116,
                              "name": "rpm_5",
                            },
                            Object {
                              "id": 117,
                              "name": "rpm_6",
                            },
                            Object {
                              "id": 118,
                              "name": "rpm_7",
                            },
                            Object {
                              "id": 119,
                              "name": "rpm_8",
                            },
                            Object {
                              "id": 120,
                              "name": "rpm_9",
                            },
                            Object {
                              "id": 121,
                              "name": "rpm_10",
                            },
                          ],
                        },
                        Object {
                          "id": 38,
                          "name": "minimal",
                          "rpms": Array [
                            Object {
                              "id": 84,
                              "name": "python2-avocado",
                            },
                          ],
                        },
                      ],
                      "repositories": Array [
                        Object {
                          "id": 1,
                          "name": "rawhide_wtih_modules",
                          "product_id": 1,
                          "product_name": "fedora",
                        },
                        Object {
                          "id": 4,
                          "name": "rawhide_wtih_modules_dup",
                          "product_id": 1,
                          "product_name": "fedora",
                        },
                      ],
                      "stream": "latest",
                      "summary": "Framework with tools and libraries for Automated Testing",
                      "uuid": "8ae7f190-0a48-41a2-93e0-7bc3e4734355",
                      "version": "20180816135607",
                    }
                  }
                />
              </Col>
            </Row>
          </TabPane>
          <TabPane
            bsClass="tab-pane"
            eventKey={2}
          >
            <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                sm={12}
              >
                <ModuleStreamDetailRepositories
                  repositories={
                    Array [
                      Object {
                        "id": 1,
                        "name": "rawhide_wtih_modules",
                        "product_id": 1,
                        "product_name": "fedora",
                      },
                      Object {
                        "id": 4,
                        "name": "rawhide_wtih_modules_dup",
                        "product_id": 1,
                        "product_name": "fedora",
                      },
                    ]
                  }
                />
              </Col>
            </Row>
          </TabPane>
          <TabPane
            bsClass="tab-pane"
            eventKey={3}
          >
            <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                sm={12}
              >
                <ModuleStreamDetailProfiles
                  profiles={
                    Array [
                      Object {
                        "id": 37,
                        "name": "default",
                        "rpms": Array [
                          Object {
                            "id": 108,
                            "name": "perl",
                          },
                          Object {
                            "id": 110,
                            "name": "foo",
                          },
                          Object {
                            "id": 111,
                            "name": "rpm_0",
                          },
                          Object {
                            "id": 112,
                            "name": "rpm_1",
                          },
                          Object {
                            "id": 113,
                            "name": "rpm_2",
                          },
                          Object {
                            "id": 114,
                            "name": "rpm_3",
                          },
                          Object {
                            "id": 115,
                            "name": "rpm_4",
                          },
                          Object {
                            "id": 116,
                            "name": "rpm_5",
                          },
                          Object {
                            "id": 117,
                            "name": "rpm_6",
                          },
                          Object {
                            "id": 118,
                            "name": "rpm_7",
                          },
                          Object {
                            "id": 119,
                            "name": "rpm_8",
                          },
                          Object {
                            "id": 120,
                            "name": "rpm_9",
                          },
                          Object {
                            "id": 121,
                            "name": "rpm_10",
                          },
                        ],
                      },
                      Object {
                        "id": 38,
                        "name": "minimal",
                        "rpms": Array [
                          Object {
                            "id": 84,
                            "name": "python2-avocado",
                          },
                        ],
                      },
                    ]
                  }
                />
              </Col>
            </Row>
          </TabPane>
          <TabPane
            bsClass="tab-pane"
            eventKey={4}
          >
            <Row
              bsClass="row"
              componentClass="div"
            >
              <Col
                bsClass="col"
                componentClass="div"
                sm={12}
              >
                <ModuleStreamDetailArtifacts
                  artifacts={
                    Array [
                      Object {
                        "id": 208,
                        "name": "python3-avocado-plugins-varianter-yaml-to-mux-0:63.0-2.module_2037+1b0ad681.noarch",
                      },
                      Object {
                        "id": 207,
                        "name": "python3-avocado-plugins-varianter-pict-0:63.0-2.module_2037+1b0ad681.noarch",
                      },
                    ]
                  }
                />
              </Col>
            </Row>
          </TabPane>
        </TabContent>
      </Grid>
    </ForwardRef>
  </LoadingState>
</div>
`;
